dbt/include/spark/macros/materializations/incremental.sql

```diff
@@ -1,24 +1,96 @@
+{% macro dbt_spark_validate_get_file_format() %}
+  {#-- Find and validate the file format #}
+  {%- set file_format = config.get("file_format", default="parquet") -%}
 
-  {%- set partitions = config.get('partition_by') -%}
-  {% if not partitions %}
-    {% do exceptions.raise_compiler_error("Table partitions are required for incremental models on Spark") %}
+  {% set invalid_file_format_msg -%}
+    Invalid file format provided: {{ file_format }}
+    Expected one of: 'text', 'csv', 'json', 'jdbc', 'parquet', 'orc', 'hive', 'delta', 'libsvm'
+  {%- endset %}
+
+  {% if file_format not in ['text', 'csv', 'json', 'jdbc', 'parquet', 'orc', 'hive', 'delta', 'libsvm'] %}
+    {% do exceptions.raise_compiler_error(invalid_file_format_msg) %}
   {% endif %}
 
-  {%- set identifier = model['alias'] -%}
-  {%- set tmp_identifier = model['alias'] ~ "__dbt_tmp" -%}
+  {% do return(file_format) %}
+{% endmacro %}
+
+{% macro dbt_spark_validate_get_incremental_strategy(file_format) %}
+  {#-- Find and validate the incremental strategy #}
+  {%- set strategy = config.get("incremental_strategy", default="insert_overwrite") -%}
+
+  {% set invalid_strategy_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    Expected one of: 'merge', 'insert_overwrite'
+  {%- endset %}
+
+  {% set invalid_merge_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    You can only choose this strategy when file_format is set to 'delta'
+  {%- endset %}
+
+  {% if strategy not in ['merge', 'insert_overwrite'] %}
+    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+  {%-else %}
+    {% if strategy == 'merge' and file_format != 'delta' %}
+      {% do exceptions.raise_compiler_error(invalid_merge_msg) %}
+    {% endif %}
+  {% endif %}
+
+  {% do return(strategy) %}
+{% endmacro %}
+
+{% macro dbt_spark_validate_merge(file_format) %}
+  {% set invalid_file_format_msg -%}
+    You can only choose the 'merge' incremental_strategy when file_format is set to 'delta'
+  {%- endset %}
+
+  {% if file_format != 'delta' %}
+    {% do exceptions.raise_compiler_error(invalid_file_format_msg) %}
+  {% endif %}
 
-  {%- set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) -%}
-  {%- set target_relation = api.Relation.create(identifier=identifier, schema=schema, database=database,  type='table') -%}
-  {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,  type='table') -%}
+{% endmacro %}
 
-  {%- set full_refresh = flags.FULL_REFRESH == True and old_relation is not none -%}
-  {%- set old_relation_is_view = old_relation is not none and old_relation.is_view -%}
 
-  {%- if full_refresh or old_relation_is_view -%}
-    {{ adapter.drop_relation(old_relation) }}
-    {%- set old_relation = none -%}
-  {%- endif %}
+{% macro dbt_spark_get_incremental_sql(strategy, source, target, unique_key) %}
+  {%- if strategy == 'insert_overwrite' -%}
```

Comment:
I split the `insert_overwrite` approach into its own macro, `get_insert_overwrite_sql`, in #60. I think you should pull out the `merge` SQL into a `get_merge_sql` macro, and then `dbt_spark_get_incremental_sql` can call `get_insert_overwrite_sql` and `get_merge_sql` based on the strategy.

A note to myself: `dbt-core` implements a common `get_merge_sql` macro [here](https://github.com/fishtown-analytics/dbt/blob/0.15.latest/core/dbt/include/global_project/macros/materializations/common/merge.sql), and the syntax for Spark is roughly identical—it looks like Spark lets you use `*` for `update set` + `insert`, where other databases require an explicit list of columns. We're changing the names of a few of these macros in the 0.16.0 release of `dbt`, so it probably makes more sense to connect this back to the core macros for the 0.16.0 release of the plugin.